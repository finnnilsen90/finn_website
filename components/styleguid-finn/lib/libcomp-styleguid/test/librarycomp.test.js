const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer');
const jest = require('jest');
const sinon = require('sinon');
let { shallow, mount, render } = require('enzyme');

const Libcomp = require('../libcomp-styleguid.js');

let newlibcompcomp = new Libcomp();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Libcomp).toBeDefined();
    })

})  
    
describe('<Libcomp />', () => {

    const wrapper = shallow(<Libcomp />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe('JSON input', () => {
    const wrapper = mount(<Libcomp />);
    const json = wrapper.state().json_menu;
    const def = wrapper.state().menu;

    it('should take in JSON file', () => {
        expect(json).toEqual(def)
    })
})