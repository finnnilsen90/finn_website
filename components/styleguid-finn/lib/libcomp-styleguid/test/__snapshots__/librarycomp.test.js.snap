// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Libcomp /> renders components 1`] = `
<div
  className="libcomp-styleguid_Container"
>
  <h2
    className="header_comp"
  >
    Library components
  </h2>
  <div
    className="hamburger_pos"
  >
    <Hamburger />
  </div>
  <div
    className="button_comp"
  >
    <button
      className="button_style_one"
    >
      Button
    </button>
    <button
      className="button_style_two"
    >
      Button
    </button>
  </div>
</div>
`;
